packages/contracts-periphery/contracts/universal/citizen-house/Badge.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { SafeTransferLib } from "@rari-capital/solmate/src/utils/SafeTransferLib.sol";
+import { ERC721 } from "@rari-capital/solmate/src/tokens/ERC721.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Badge contract
+ * @notice A minimalist soulbound ERC-721 Badge implementation
+ * @author OPTIMISM + GITCOIN
+ */
+contract Badge is ERC721, Ownable {
+    /**
+     * @notice adminContract address i.e BadgeAdmin address
+     */
+    address public adminContract;
+
+    /**
+     * @notice BaseURI of the NFT
+     */
+    string private baseURI;
```

Comment:
This can be immutable

Reply:
As per documentation here: https://docs.soliditylang.org/en/v0.8.15/contracts.html?highlight=immutable#constant-and-immutable-state-variables `string` only supports `constant` and must provided at compile-time. 
Alternatively, we can hash the `baseURI` for use with `immutable`. Or we could just allow an admin to update the `baseURI`.

Reply:
I went with adding a `updateBaseURI(string) onlyOwner` for now 👍